primitives/header-chain/src/lib.rs

```diff
@@ -110,3 +112,107 @@ impl<H, P> AncestryChecker<H, P> for () {
 		true
 	}
 }
+
+/// A simple ancestry checker which verifies ancestry by walking every header between `child` and
+/// `ancestor`.
+pub struct LinearAncestryChecker;
+
+impl<H: HeaderT> AncestryChecker<H, Vec<H>> for LinearAncestryChecker {
+	fn are_ancestors(ancestor: &H, child: &H, proof: &Vec<H>) -> bool {
+		// You can't be your own parent
+		if ancestor == child {
+			return false;
+		}
+
+		// Let's make sure that the given headers are actually in the proof
+		if let Some(first) = proof.first() {
+			if first != ancestor {
+				return false;
+			}
+		} else {
+			return false;
+		};
+
+		if let Some(last) = proof.last() {
+			if last != child {
+				return false;
+			}
+		} else {
+			return false;
+		};
```

Comment:
```suggestion
		match proof.last() {
		    Some(last) if last == child => {},
		    _ => return false,
		}
```
likewise

Reply:
I knew there was a more idiomatic way to write this, but couldn't see it yesterday